# <b><center> Social Networks EDA </b></center>


## Entendimiento de los Datos:

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import pathlib
import re
from difflib import SequenceMatcher

## <u>Leer los datos que en el nombre contienen un mes:</u></br>
### <b>Instagram:</b>

Cargar los 12 archivos que en el nombre tienen un mes, en 3  DataFrames, uno de cada red social, donde cada uno de los cuales debe contener las siguientes columnas:

Instagram:
* Username
* Name
* Subscribers o Followers
* Country
* Authentic
* Engagement
* Category1
* Category2
* Month

In [2]:
DATA_DIR = pathlib.Path().resolve().joinpath('DATA')
ig = dict()
set_months = {'June', 'September', 'November'}
for path in DATA_DIR.glob('i*2022.csv'):
    target_month = re.search(r'instagram(.*)2022', str(path), re.IGNORECASE).group(1)
    real_month = max(set_months, key = lambda month: SequenceMatcher(None, target_month, month).ratio())
    ig.update({
        'ig_' + real_month:
        pd.read_csv(path)
    })

In [3]:
for name, df in ig.items():
    print(name)
    print(df.head())

ig_November
   s.no          Name     Instagram Name                 Category-1 Followers  \
0     1     cristiano  Cristiano Ronaldo         Sports with a ball    483.1M   
1     2   kyliejenner            Kylie 🤍                    Fashion      370M   
2     3      leomessi          Leo Messi         Sports with a ball    357.8M   
3     4  arianagrande      Ariana Grande                      Music    331.6M   
4     5       zendaya            Zendaya  Cinema & Actors/actresses    152.2M   

       \nCountry Eng. (Auth.) Eng. (Avg.) Category-2  
0          India         4.5M        5.6M        NaN  
1  United States         2.8M        4.4M   Modeling  
2      Indonesia         3.2M        4.1M     Family  
3  United States         2.9M          4M        NaN  
4  United States         4.8M        6.5M    Fashion  
ig_June
  instagram name influencer name           Category_1   Category_2  \
0            433              433  Sports with a ball          NaN   
1  ____kimwoobin       

In [4]:
ig_june = pd.read_csv('DATA/instagramjune2022.csv')
ig_june = (
    ig_june
    .assign(Month='June')
    .rename(columns={'Subscribers count': 'Suscribers', 'influencer name': 'Name'})
    )

ig_june.info()
ig_june.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1022 entries, 0 to 1021
Data columns (total 9 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   instagram name    1022 non-null   object
 1   influencer name   996 non-null    object
 2   Category_1        927 non-null    object
 3   Category_2        298 non-null    object
 4   Suscribers        1022 non-null   object
 5   Views avg.        1022 non-null   object
 6   Likes avg         1022 non-null   object
 7   Comments avg.     1006 non-null   object
 8   Month             1022 non-null   object
dtypes: object(9)
memory usage: 72.0+ KB


instagram name influencer name           Category_1   Category_2 Suscribers  \
0            433              433  Sports with a ball          NaN      50.7M   
1  ____kimwoobin         김우빈           Lifestyle          NaN       3.6M   
2   __youngbae__          TAEYANG               Music          NaN      13.4M   
3    _agentgirl_    НАСТЯ ИВЛЕЕВА               Shows          NaN      18.6M   
4  _hakkencoser_        Hakken 八犬           Lifestyle  Photography       3.5M   

      Views avg. Likes avg Comments avg. Month  
0          Spain    274.9K        456.7K  June  
1      Indonesia    449.4K        592.7K  June  
2      Indonesia    269.6K        304.1K  June  
3         Russia    164.4K        189.4K  June  
4  United States    525.1K          643K  June

In [5]:
ig_sep = pd.read_csv('DATA/instagramsep2022.csv')
ig_sep = (
    ig_sep
    .assign(Month='September')
    .rename(columns={'Instagram name': 'instagram name', 'Audience country': 'Views avg.'})
    .drop('S.no', axis=1)
    )
ig_sep.info()
ig_sep.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1050 entries, 0 to 1049
Data columns (total 9 columns):
 #   Column                 Non-Null Count  Dtype 
---  ------                 --------------  ----- 
 0   instagram name         1050 non-null   object
 1    Name                  1029 non-null   object
 2   Subscribers            1050 non-null   object
 3   Views avg.             1046 non-null   object
 4   Authentic engagement
  1050 non-null   object
 5   Engagement average
   1050 non-null   object
 6   Category_1             802 non-null    object
 7   Category_2             250 non-null    object
 8   Month                  1050 non-null   object
dtypes: object(9)
memory usage: 74.0+ KB


instagram name               Name Subscribers     Views avg.  \
0      cristiano  Cristiano Ronaldo      477.9M          India   
1    kyliejenner            Kylie 🤍      368.1M  United States   
2   arianagrande      Ariana Grande      329.6M  United States   
3       leomessi          Leo Messi      358.6M      Indonesia   
4        zendaya            Zendaya      151.1M  United States   

  Authentic engagement\n Engagement average\r\n                 Category_1  \
0                     5M                   6.2M         Sports with a ball   
1                   3.5M                   5.5M                    Fashion   
2                   2.9M                     4M                        NaN   
3                   2.7M                   3.5M         Sports with a ball   
4                   4.3M                   5.8M  Cinema & Actors/actresses   

  Category_2      Month  
0        NaN  September  
1   Modeling  September  
2        NaN  September  
3     Family  September  
4    Fashion  September

In [6]:
ig_nov = pd.read_csv('DATA/instagramNov2022.csv')
ig_nov = ig_nov.assign(Month='November')
ig_nov.drop('s.no', inplace=True, axis=1)
ig_nov.info()
ig_nov.head()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000 entries, 0 to 999
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   Name            1000 non-null   object
 1   Instagram Name  978 non-null    object
 2   Category-1      952 non-null    object
 3   Followers       1000 non-null   object
 4   
Country        996 non-null    object
 5   Eng. (Auth.)    1000 non-null   object
 6   Eng. (Avg.)     1000 non-null   object
 7   Category-2      339 non-null    object
 8   Month           1000 non-null   object
dtypes: object(9)
memory usage: 70.4+ KB


Name     Instagram Name                 Category-1 Followers  \
0     cristiano  Cristiano Ronaldo         Sports with a ball    483.1M   
1   kyliejenner            Kylie 🤍                    Fashion      370M   
2      leomessi          Leo Messi         Sports with a ball    357.8M   
3  arianagrande      Ariana Grande                      Music    331.6M   
4       zendaya            Zendaya  Cinema & Actors/actresses    152.2M   

       \nCountry Eng. (Auth.) Eng. (Avg.) Category-2     Month  
0          India         4.5M        5.6M        NaN  November  
1  United States         2.8M        4.4M   Modeling  November  
2      Indonesia         3.2M        4.1M     Family  November  
3  United States         2.9M          4M        NaN  November  
4  United States         4.8M        6.5M    Fashion  November

In [117]:
ig_dec= pd.read_csv('DATA/INSTAGRAMDEC2022.csv')
ig_dec = ig_dec.assign(Month='December')
ig_dec.head()

Rank           name     instagram name          Category_1 Category_2  \
0     1       leomessi          Leo Messi  Sports with a ball     Family   
1     2      cristiano  Cristiano Ronaldo  Sports with a ball        NaN   
2     3       neymarjr              NJ 🇧🇷  Sports with a ball        NaN   
3     4    kyliejenner            Kylie 🤍             Fashion   Modeling   
4     5  kendalljenner            Kendall            Modeling    Fashion   

  followers        country Eng. (Auth.) Eng. (Avg.)     Month  
0    409.8M      Argentina        17.8M       23.4M  December  
1      523M          India        11.7M       14.7M  December  
2    198.9M         Brazil         6.7M        9.5M  December  
3    376.3M  United States         2.7M        4.3M  December  
4      268M  United States         3.2M          5M  December

In [118]:
ig_dec = ig_dec.rename(columns = {"name": "name",
                       "instagram name": "username",
                       "Category_1": "category1",
                       "Category_2": "category2",
                       "Eng. (Auth.)": "authentic",
                       "Eng. (Avg.)": "average",
                       "Month": "month"
                       })
# 


In [119]:
ig_nov = ig_nov.rename(columns={"Name": "name",
                               "Instagram Name": "username",
                               "Category-1": "category1",
                               "Followers": "followers",
                               "\nCountry": "country",
                               "Eng. (Auth.)": "authentic",
                               "Eng. (Avg.)": "average",
                               "Category-2": "category2",
                               "Month": "month"})

In [120]:
ig_sep = ig_sep.rename(columns={"instagram name": "username",
                               " Name": "name",
                               "Subscribers": "followers",
                               "Views avg.": "country",
                               "Authentic engagement\n": "authentic",
                                "Engagement average\r\n": "average",
                                "Category_1": "category1",
                                "Category_2": "category2",
                                "Month": "month"})


In [121]:
ig_june = ig_june.rename(columns={"instagram name": "username",
                                 "influencer name ":"name",
                                 "Category_1": "category1",
                                 "Category_2": "category2",
                                 "Suscribers": "followers",
                                 "Views avg.": "country",
                                 "Likes avg": "authentic",
                                 "Comments avg.": "average",
                                 "Month": "month"})


In [122]:
ig_months = pd.concat([ig_june, ig_sep, ig_nov, ig_dec], ignore_index=True).drop_duplicates(subset=["username"], keep='first')
ig_months.drop("Rank", inplace=True, axis=1)
ig_months

username             name           category1  \
0                     433              433  Sports with a ball   
1           ____kimwoobin         김우빈           Lifestyle   
2            __youngbae__          TAEYANG               Music   
3             _agentgirl_    НАСТЯ ИВЛЕЕВА               Shows   
4           _hakkencoser_        Hakken 八犬           Lifestyle   
...                   ...              ...                 ...   
4063                 nova     strangehuman           Lifestyle   
4064  HYBE LABELS Updates      labels.hybe               Shows   
4065   FARGO’S CRAZY LIFE        torylanez               Shows   
4069                   이찬        feat.dino           Lifestyle   
4071  Ravindrasinh jadeja  ravindra.jadeja           Lifestyle   

               category2 followers        country authentic average     month  
0                    NaN  50700000          Spain    274.9K  456.7K      June  
1                    NaN   3600000      Indonesia    449.4K  592.7K      June  
2                    NaN  13400000      Indonesia    269.6K  304.1K      June  
3                    NaN  18600000         Russia    164.4K  189.4K      June  
4            Photography   3500000  United States    525.1K    643K      June  
...                  ...       ...            ...       ...     ...       ...  
4063                 NaN      4.6M  United States    371.8K  515.6K  December  
4064               Music      2.3M          India    891.1K      1M  December  
4065               Music     11.9M  United States     85.1K    196K  December  
4069                 NaN      2.7M    South Korea    716.6K  876.4K  December  
4071  Sports with a ball      4.8M          India    388.9K  485.1K  December  

[2409 rows x 9 columns]

In [123]:
ig_months[ig_months.duplicated() == True].size

0

ig_months es la combinación de todos los meses, eliminando aquellos registros duplicados.
Tener cuidado con los datos numéricos, ya que son datos incompletos. Para sacar promedios se tiene que considerar los df por separado

### TikTok:

In [124]:
tt_june= pd.read_csv('DATA/tiktokjune2022.csv')
tt_june = tt_june.assign(Month='June')
tt_june

Tiktoker         influencer name Subscribers count Views avg.  \
0       ___nicoichi___           NICO【平成フラミンゴ】              1.3M       1.9M   
1      _.anime.edits_0                   video               97K       1.7M   
2     _.respect.boss._  Respect compilation 🤝🥶              1.5M       4.4M   
3     _.the.meme.hub._            RANDOM MEMES              2.3M       3.8M   
4       _andrewcurtiss           Andrew Curtis              2.5M       1.4M   
...                ...                     ...               ...        ...   
1002              zeth                    Zeth             12.5M       6.7M   
1003  zodiac.boyfriend      Zodiac Boyfriend🪐🔮              1.8M       1.9M   
1004       zowloficial               Lord Zowl              1.1M       1.4M   
1005          zrt.foot                     ZR7            100.8K       2.3M   
1006         zubarefff       Зубарев Александр                4M       1.9M   

     Likes avg Comments avg. Shares avg Month  
0       225.3K          1.7K       1.1K  June  
1        75.6K          1.5K       8.7K  June  
2       241.9K           249        797  June  
3       787.7K          2.9K         9K  June  
4       300.4K          5.8K         4K  June  
...        ...           ...        ...   ...  
1002      1.3M            3K       7.7K  June  
1003    272.7K          6.8K       3.6K  June  
1004    178.1K          1.4K       2.1K  June  
1005      218K          1.7K       6.4K  June  
1006      249K          1.3K       1.6K  June  

[1007 rows x 8 columns]

In [125]:
tt_sep= pd.read_csv('DATA/Tiktoksep2022.csv')
tt_sep = tt_sep.assign(Month='September')
tt_sep

S.no   Tiktoker name    Tiktok name Subscribers Views avg. Likes avg.  \
0       1   jypestraykids     Stray Kids       13.8M       6.4M       2.3M   
1       2      khaby.lame   Khabane lame      149.2M      17.3M       2.3M   
2       3  scarlettsspam2       scarlett        2.1M      17.9M     845.8K   
3       4       addisonre    Addison Rae       88.7M        22M     906.6K   
4       5      belindatok        Belinda        4.8M      14.2M       1.5M   
..    ...             ...            ...         ...        ...        ...   
995   996    brendadialoy  Brenda Dialoy      371.9K       1.2M     187.5K   
996   997     jujufitcats   Juju Fitcats        4.3M       2.2M     280.8K   
997   998          xoteam        XO Team       37.8M       2.5M     180.2K   
998   999        kimsnwuo             ️️      100.9K     957.4K     226.8K   
999  1000        jass_mcg       jass_mcg       52.2K     503.8K       116K   

    Comments avg. Shares avg.      Month  
0           50.2K       34.2K  September  
1           15.2K        8.7K  September  
2           53.9K        6.3K  September  
3            7.6K       26.2K  September  
4           14.5K       15.3K  September  
..            ...         ...        ...  
995           626          2K  September  
996           380         558  September  
997           859         450  September  
998           908        1.7K  September  
999          8.1K        1.8K  September  

[1000 rows x 9 columns]

In [126]:
tt_nov= pd.read_csv('DATA/tiktoknov2022.csv')
tt_nov = tt_nov.assign(Month='November')
tt_nov

row-cell    Tiktoker name       Tiktok name \nFollowers Views (Avg.)  \
0          NaN              NaN               NaN         NaN          NaN   
1          1.0      yossikramer             Yossi      145.6K        26.8M   
2          2.0          mrbeast           MrBeast       55.6M        38.1M   
3          3.0         adinross              adin        5.4M        16.6M   
4          4.0   daniel.labelle    Daniel LaBelle       27.9M        11.8M   
...        ...              ...               ...         ...          ...   
996      996.0  nathanmccarron0      Nathan’s Art        1.6M         1.1M   
997      997.0        lacha21.1   Lazaro Gonzalez      669.4K       750.3K   
998      998.0          hyoga.x             Hyoga        552K           1M   
999      999.0   rrqr7tatsumaki    Rivaldi Fataah        1.1M           2M   
1000    1000.0         _catben_  Catherine Benson       10.9M         1.6M   

     Likes (Avg.) Comments (Avg.) Shares (Avg.)     Month  
0             NaN             NaN           NaN  November  
1            2.2M           30.1K         71.2K  November  
2            3.8M             19K          7.8K  November  
3              2M           10.9K         12.5K  November  
4            1.7M            6.9K         29.3K  November  
...           ...             ...           ...       ...  
996        117.6K            1.4K          1.1K  November  
997        118.3K             655            3K  November  
998        121.9K            1.9K           727  November  
999        112.5K              1K           414  November  
1000       212.5K             665           318  November  

[1001 rows x 9 columns]

In [127]:
tt_dec= pd.read_csv('DATA/TIKTOKDEC2022.csv')
tt_dec = tt_dec.assign(Month='December')
tt_dec

Rank Tiktoker name                     Tiktok name followers views(avg)  \
0       1       mrbeast                         MrBeast     60.3M      29.2M   
1       2        karolg                         Karol G     42.4M      23.7M   
2       3         yzn47                      يزن الأسمر      8.9M      48.9M   
3       4    centralcee                      CentralCee      4.4M      19.8M   
4       5      adinross                            adin      6.1M      21.1M   
..    ...           ...                             ...       ...        ...   
995   996   jiembasands                    Jiemba Sands      4.9M       3.4M   
996   997    crissa_ace                  Crissa Jackson     14.9M       1.6M   
997   998      ichadude                   Alyssa & Dude    468.8K       2.3M   
998   999     kanebrown                      Kane Brown      5.2M       1.7M   
999  1000      nnennab_  Nnenna B | NYC Creator & Actor    149.1K     606.2K   

    likes(avg.) comments(avg.) shares(avg.)     Month  
0          3.5M          30.8K         7.2K  December  
1          3.4M          21.7K        25.7K  December  
2        998.4K          16.3K        60.9K  December  
3          3.6M          23.3K        24.2K  December  
4          3.3M          17.5K        25.3K  December  
..          ...            ...          ...       ...  
995      247.4K            207          264  December  
996      141.7K           1.2K          580  December  
997       85.5K            997           1K  December  
998       96.7K           1.3K         1.2K  December  
999       79.5K           2.1K         6.1K  December  

[1000 rows x 9 columns]

### Youtube:

In [128]:
y_june= pd.read_csv('DATA/youtubejune2022.csv')
y_june = y_june.assign(Month='June')
y_june

channel name  \
0                                        juptube   
1     Baby Big Mouth Kids - Sing, Dance, Explore   
2                                shfa show India   
3           Blippi - Educational Videos for Kids   
4                                 infobellstamil   
...                                          ...   
1016                                Power Vision   
1017                                 POPSVIETNAM   
1018                               Niki and Gabi   
1019                                    Corridor   
1020                                     JoshDub   

                                        youTuber       Category Category_2  \
0                             Galinha Pintadinha  Music & Dance        NaN   
1     Baby Big Mouth Kids - Sing, Dance, Explore      Animation       Toys   
2                                shfa show India    Daily vlogs        NaN   
3           Blippi - Educational Videos for Kids      Education       Toys   
4                              infobells - Tamil      Education        NaN   
...                                          ...            ...        ...   
1016                                Power Vision            NaN        NaN   
1017                                  POPS MUSIC  Music & Dance        NaN   
1018                               Niki and Gabi        Fashion      Humor   
1019                                    Corridor      Animation        NaN   
1020                                     JoshDub      Animation      Humor   

     Subscribers count        Country Views avg. Likes avg Comments avg. Month  
0                30.5M              -          0      N/A'          N/A'  June  
1                10.6M              -      348.5         3          N/A'  June  
2                18.9M              -       1.2M      8.7K          N/A'  June  
3                15.7M              -       1.2M      4.2K          N/A'  June  
4                26.5M              -       1.3M      5.8K          N/A'  June  
...                ...            ...        ...       ...           ...   ...  
1016              9.5M  United States     137.3K      1.3K            39  June  
1017              9.5M  United States       4.6K       144             5  June  
1018              9.5M  United States     387.6K     17.8K           516  June  
1019              9.5M  United States       1.9M    141.6K          4.7K  June  
1020              9.5M  United States     713.6K     35.4K          1.2K  June  

[1021 rows x 10 columns]

In [129]:
y_sep= pd.read_csv('DATA/Youtubesep2022.csv')
y_sep = y_sep.assign(Month='September')
y_sep.head()

S.no                        Name           Youtuber        Country  \
0     1                    T-Series            tseries          India   
1     2  Cocomelon - Nursery Rhymes          checkgate            NaN   
2     3                   PewDiePie          PewDiePie  United States   
3     4                     MrBeast        MrBeast6000  United States   
4     5           ✿ Kids Diana Show  ✿ Kids Diana Show            NaN   

   Subscribers     Category_2 Avg. views\r\n Avg. likes Avg Comments  \
0       224.4M  Music & Dance         141.5K       4.6K          125   
1       142.5M      Education          12.5M      64.8K            0   
2       111.6M      Animation           1.9M     130.5K         7.4K   
3       104.1M    Video games          44.8M       2.2M        70.9K   
4       100.5M      Animation           6.9M      34.8K            0   

  Category_3      Month  
0        NaN  September  
1        NaN  September  
2        NaN  September  
3      Humor  September  
4       Toys  September

In [130]:
y_nov= pd.read_csv('DATA/youtubenov2022.csv')
y_nov = y_nov.assign(Month='November')
y_nov

s.no    Youtube channel               youtuber name         Category  \
0       1            tseries                    T-Series    Music & Dance   
1       2          checkgate  Cocomelon - Nursery Rhymes        Education   
2       3          PewDiePie                   PewDiePie        Animation   
3       4        MrBeast6000                     MrBeast      Video games   
4       5  ✿ Kids Diana Show           ✿ Kids Diana Show        Animation   
..    ...                ...                         ...              ...   
995   996               RUOK                        RUOK              NaN   
996   997     FoxNewsChannel                    Fox News  News & Politics   
997   998   SULE PRODUCTIONS            SULE PRODUCTIONS              NaN   
998   999            MBCkpop                     MBCkpop    Music & Dance   
999  1000      Dynamo Gaming               Dynamo Gaming              NaN   

    Followers        Country \nViews (Avg.) Likes (Avg.) Comments (Avg.)  \
0      229.5M          India         103.8K         2.8K             158   
1      146.6M            NaN           9.5M        48.3K               0   
2      111.6M  United States           1.4M       110.7K            7.8K   
3      109.1M  United States          38.4M         3.1M           56.7K   
4      102.5M            NaN           2.4M        15.7K               0   
..        ...            ...            ...          ...             ...   
995     10.1M         Brazil         215.4K        25.2K            1.3K   
996     10.1M  United States          98.4K           4K             884   
997     10.1M      Indonesia         218.3K         4.5K             306   
998     10.1M       Malaysia          19.7K         2.6K              92   
999     10.1M          India         139.6K        15.4K             125   

    Category-2     Month  
0          NaN  November  
1          NaN  November  
2          NaN  November  
3        Humor  November  
4         Toys  November  
..         ...       ...  
995        NaN  November  
996        NaN  November  
997        NaN  November  
998        NaN  November  
999        NaN  November  

[1000 rows x 11 columns]

In [131]:
y_dec= pd.read_csv('DATA/YOUTUBEDEC2022.csv')
y_dec = y_dec.assign(Month='December')
y_dec

s.no            Youtube channel               youtuber name  \
0        1                    tseries                    T-Series   
1        2                   setindia                   SET India   
2        3                  checkgate  Cocomelon - Nursery Rhymes   
3        4                    MrBeast                     MrBeast   
4        5                  PewDiePie                   PewDiePie   
...    ...                        ...                         ...   
998    999              ClassicMrBean             Classic Mr Bean   
999   1000         Anupam Movie Songs          Anupam Movie Songs   
1000   970                 AnandAudio                 Anand Audio   
1001   971                   Wylsacom                    Wylsacom   
1002   986  Eagle Home Entertainments   Eagle Home Entertainments   

                  Category Followers        Country \nViews (Avg.)  \
0            Music & Dance    229.5M          India         103.8K   
1                      NaN    147.2M          India              0   
2                Education    146.6M            NaN           9.5M   
3              Video games      117M  United States          68.5M   
4                Animation    111.6M  United States           1.4M   
...                    ...       ...            ...            ...   
998                    NaN     10.1M          India          48.3K   
999          Music & Dance     10.1M     Bangladesh          41.2K   
1000                   NaN     10.3M          India         175.5K   
1001  Science & Technology     10.3M         Russia         291.7K   
1002                   NaN     10.2M          India            724   

     Likes (Avg.) Comments (Avg.)       Category-2     Month  
0            2.8K             158              NaN  December  
1               0               0              NaN  December  
2           48.3K               0              NaN  December  
3            2.9M           99.9K            Humor  December  
4          110.7K            7.8K              NaN  December  
...           ...             ...              ...       ...  
998           624              23              NaN  December  
999           315              18              NaN  December  
1000         2.2K             109              NaN  December  
1001          13K             950  News & Politics  December  
1002            3               0              NaN  December  

[1003 rows x 11 columns]

TikTok:
* Username
* Name
* Followers o Suscribers
* Views
* Likes
* Comments
* Shares
* Month

Youtube:
* Username
* Name
* Country
* Suscribers o Folowers
* Views
* Likes
* Comments
* Category1
* Category2
* Month

# Análisis instragam

In [132]:
print(ig_months.shape)
ig_months.head()

(2409, 9)


username           name           category1    category2 followers  \
0            433            433  Sports with a ball          NaN  50700000   
1  ____kimwoobin       김우빈           Lifestyle          NaN   3600000   
2   __youngbae__        TAEYANG               Music          NaN  13400000   
3    _agentgirl_  НАСТЯ ИВЛЕЕВА               Shows          NaN  18600000   
4  _hakkencoser_      Hakken 八犬           Lifestyle  Photography   3500000   

         country authentic average month  
0          Spain    274.9K  456.7K  June  
1      Indonesia    449.4K  592.7K  June  
2      Indonesia    269.6K  304.1K  June  
3         Russia    164.4K  189.4K  June  
4  United States    525.1K    643K  June

### Número total de cuentas distintas en instagram: 2409

### Media de seguidores
Si un registro se encuentra en más de 1 mes, promediar todas las cantidades de seguidores de todos los df.

Vamos a agarrar como pivote el username del dataframe combinado

In [133]:
ig_dec["followers"] = ig_dec["followers"].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)


In [134]:
ig_nov["followers"] = ig_nov["followers"].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)


In [135]:
ig_sep["followers"] = ig_sep["followers"].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)


In [136]:
ig_june["followers"] = ig_june["followers"].replace({'K': '*1e3', 'M': '*1e6', 'G':'*1e9'}, regex=True).map(pd.eval).astype(int)


In [152]:
temp_dec = ig_dec.loc[:, ["username","followers"]]
temp_nov = ig_nov.loc[:, ["username", "followers"]]
temp_sep = ig_sep.loc[:, ["username", "followers"]]
temp_june = ig_june.loc[:, ["username", "followers"]]
dataframes = {
                'diciembre': temp_dec,
                 'noviembre': temp_nov,
                 'septiembre': temp_sep,
                 'junio': temp_june
            }


In [153]:
for user in ig_months["username"]:
    suma_dataframes = 0
    suma_seguidores = 0
    for key, value in dataframes.items():
        if user in value["username"].values:
            # Se encuentra en más de 1
            #print(f'{user} se encuentra en {key} con total de seguidores {value.loc[value["username"] == user, "followers"]}')
            suma_seguidores += np.sum(value.loc[value["username"] == user, "followers"])
            suma_dataframes += 1
        else:
            # Se encuentra solo en 1
            suma_dataframes = 1
    print(f'Media de los seguidores de {user}: {suma_seguidores / suma_dataframes }')
            

Media de los seguidores de 433: 53075000.0
Media de los seguidores de ____kimwoobin: 1800000.0
Media de los seguidores de __youngbae__: 9000000.0
Media de los seguidores de _agentgirl_: 9300000.0
Media de los seguidores de _hakkencoser_: 3566666.6666666665
Media de los seguidores de _imyour_joy: 9400000.0
Media de los seguidores de _jeongjaehyun: 8100000.0
Media de los seguidores de _mariahwasa: 5499999.666666667
Media de los seguidores de _rl9: 18666666.666666668
Media de los seguidores de _seorina: 4100000.0
Media de los seguidores de _shotaroo_: 2650000.0
Media de los seguidores de _zyozyo: 3433333.3333333335
Media de los seguidores de 03_hu: 2100000.0
Media de los seguidores de 21savage: 9033333.333333334
Media de los seguidores de 39saku_chan: 2600000.0
Media de los seguidores de 3gerardpique: 10200000.0
Media de los seguidores de 5sos: 5050000.0
Media de los seguidores de 9gag: 38766666.666666664
Media de los seguidores de ab: 4800000.0
Media de los seguidores de abdevilliers17: 